genesyscloud/journey_views/resource_genesyscloud_journey_views_utils.go

```diff
@@ -0,0 +1,253 @@
+package journey_views
+
+import (
+	"errors"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v130/platformclientv2"
+	"log"
+	"terraform-provider-genesyscloud/genesyscloud/util/resourcedata"
+)
+
+func buildElements(d *schema.ResourceData) (*[]platformclientv2.Journeyviewelement, error) {
+	elementsSlice := d.Get("elements").([]interface{})
+	if len(elementsSlice) == 0 {
+		emptySlice := make([]platformclientv2.Journeyviewelement, 0)
+		return &emptySlice, nil
+	}
+
+	var elements []platformclientv2.Journeyviewelement
+
+	for _, elem := range elementsSlice {
+		elemMap, ok := elem.(map[string]interface{})
+		if !ok {
+			return nil, errors.New("element is not a map[string]interface{}")
+		}
+
+		var element platformclientv2.Journeyviewelement
+		element.Id = getStringPointerFromInterface(elemMap["id"])
+		element.Name = getStringPointerFromInterface(elemMap["name"])
+
+		if attributesSlice, ok := elemMap["attributes"].([]interface{}); ok {
+			attributes := parseJourneyViewElementAttributes(attributesSlice)
+			element.Attributes = &attributes
+		}
+
+		if filterSlice, ok := elemMap["filter"].([]interface{}); ok {
+			filter := parseJourneyViewElementFilter(filterSlice)
+			element.Filter = &filter
+		}
+
+		if followedBySlice, ok := elemMap["followed_by"].([]interface{}); ok {
+			followedBy := make([]platformclientv2.Journeyviewlink, len(followedBySlice))
+			for i, fb := range followedBySlice {
+				followedByMap, ok := fb.(map[string]interface{})
+				if !ok {
+					return nil, errors.New("followedBy element is not a map[string]interface{}")
+				}
+				followedBy[i] = parseJourneyViewLink(followedByMap)
+			}
+			element.FollowedBy = &followedBy
+		}
+		elements = append(elements, element)
+	}
+
+	return &elements, nil
+}
+
+func parseJourneyViewElementAttributes(attributesSlice []interface{}) platformclientv2.Journeyviewelementattributes {
```

Comment:
Instead of parse we tend to use build for this type of functions, so this would be `buildJourneyViewElementAttributes`, same for the functions below